client/app/components/queries/query-editor.js

```diff
@@ -119,31 +121,43 @@ function queryEditor(QuerySnippet, $timeout) {
 
         const schemaCompleter = {
           getCompletions(state, session, pos, prefix, callback) {
-            if (prefix.length === 0 || !$scope.schema) {
+            // make a variable for the auto completion in the query editor
+            $scope.autoCompleteSchema = $scope.schema; // removeExtraSchemaInfo(
+            if (prefix.length === 0 || !$scope.autoCompleteSchema) {
               callback(null, []);
               return;
             }
 
-            if (!$scope.schema.keywords) {
+            if (!$scope.autoCompleteSchema.keywords) {
               const keywords = {};
 
-              $scope.schema.forEach((table) => {
+              $scope.autoCompleteSchema.forEach((table) => {
                 keywords[table.name] = 'Table';
 
-                table.columns.forEach((c) => {
-                  keywords[c] = 'Column';
+                table.columns.forEach((c) => { // autoCompleteColumns
+                  if (c.charAt(c.length - 1) === ')') {
+                    let parensStartAt = c.indexOf('(') - 1;
+                    c = c.substring(0, parensStartAt);
+                    parensStartAt = 1; // linter complains without this line
+                  }
+                  // remove '[P] ' for partition keys
+                  if (c.charAt(0) === '[') {
```

Comment:
I don't think we have [P] in this code base yet.

Reply:
Correct, but that is a pending PR and because it's an `if` statement it doesn't hurt anything prior to that PR being merged.

Reply:
Hm, I see, I'm afraid this will need to be removed for now since I got feedback from Arik about #2764 being problematic in terms of UI design. I will follow-up on this in the PR there and ask the Redash UX people to weigh in on a way forward. In the meantime we shouldn't hold back this PR here.